File: src/main/java/javabot/web/views/IndexView.java
package javabot.web.views;

import com.google.inject.Injector;

import javax.servlet.http.HttpServletRequest;
import javax.ws.rs.WebApplicationException;
import java.io.IOException;

public class IndexView extends MainView {
    public IndexView(final Injector injector, final HttpServletRequest request) {
        super(injector, request);
    }

    @Override
    public String getChildView() {
        return "/index.ftl";
    }
}

File: src/main/kotlin/javabot/kotlin/web/views/IndexView.kt
package javabot.kotlin.web.views

import com.google.inject.Inject
import com.google.inject.assistedinject.Assisted
import javabot.dao.AdminDao
import javabot.dao.ChannelDao
import javabot.dao.FactoidDao
import javax.servlet.http.HttpServletRequest

class IndexView @Inject constructor(
        adminDao: AdminDao,
        channelDao: ChannelDao,
        factoidDao: FactoidDao,
        @Assisted request: HttpServletRequest) : MainView(adminDao, channelDao, factoidDao, request) {

    override fun getChildView(): String {
        return "/index.ftl"
    }
}
